server/src/com/thoughtworks/go/config/update/TemplateConfigCommand.java

```diff
@@ -46,7 +46,7 @@ public TemplateConfigCommand(PipelineTemplateConfig templateConfig, LocalizedOpe
     public boolean isValid(CruiseConfig preprocessedConfig) {
         TemplatesConfig templatesConfig = preprocessedConfig.getTemplates();
         preprocessedTemplateConfig = findAddedTemplate(preprocessedConfig);
-        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(preprocessedConfig));
+        preprocessedTemplateConfig.validate(ConfigSaveValidationContext.forChain(templatesConfig));
```

Comment:
I know we talked about this. Since there is a much smaller part of the config being sent here, are you sure there is no other part of the config which was being validated here that isn't any more? Are you depending on the preprocessed config being validated later somewhere?